config/puma.rb

```diff
@@ -1,16 +1,40 @@
-# Puma can serve each request in a thread from an internal thread pool.
-# The `threads` method setting takes two numbers: a minimum and maximum.
-# Any libraries that use thread pools should be configured to match
-# the maximum value specified for Puma. Default is set to 5 threads for minimum
-# and maximum; this matches the default thread size of Active Record.
+# This configuration file will be evaluated by Puma. The top-level methods that
+# are invoked here are part of Puma's configuration DSL. For more information
+# about methods provided by the DSL, see https://puma.io/puma/Puma/DSL.html.
+
+# Puma starts a configurable number of processes (workers) and each process
+# serves each request in a thread from an internal thread pool.
+#
+# The ideal number of threads per worker depends both on how much time the
+# application spends waiting for IO operations and on how much you wish to
+# to prioritize throughput over latency.
+#
+# As a rule of thumb, increasing the number of threads will increase how much
+# traffic a given process can handle (throughput), but due to CRuby's
+# Global VM Lock (GVL) it has diminishing returns and will degrade the
+# response time (latency) of the application.
+#
+# The default is set to 3 threads as it's deemed a decent compromise between
+# throughput and latency for the average Rails application.
 #
+# Any libraries that use a connection pool or another resource pool should
+# be configured to provide at least as many connections as the number of
+# threads. This includes Active Record's `pool` parameter in `database.yml`.
 threads_count = ENV.fetch('RAILS_MAX_THREADS', 3)
 threads(threads_count, threads_count)
 
 # Specifies the `port` that Puma will listen on to receive requests; default is 3000.
 #
 port(ENV.fetch('PORT', 3000))
 
+# Allow puma to be restarted by `rails restart` command.
+plugin(:tmp_restart)
+
+# Only use a pidfile when requested
+if ENV['PIDFILE']
+  pidfile(ENV['PIDFILE'])
+end
```

Comment:
I'm not sure quite what this is about; might be worth looking into, but not a blocker, I think, since it doesn't seem necessary / hugely important.

Reply:
Ticket: https://davidrunger.atlassian.net/browse/DEV-87